File: polyglot/leaderboard_sort.nim
#!/usr/bin/env nim r

# Leaderboard sorting (highest balance first).
# Input format: lines like "username creditsMinor"

import std/[strutils, sequtils, algorithm]

type Row = object
  username: string
  creditsMinor: int

var rows: seq[Row] = @[]
for line in stdin.lines:
  let s = line.strip
  if s.len == 0: continue
  let parts = s.splitWhitespace
  if parts.len < 2: continue
  rows.add Row(username: parts[0], creditsMinor: parseInt(parts[1]))

rows.sort(proc(a, b: Row): int = cmp(b.creditsMinor, a.creditsMinor))

for i, r in rows:
  echo (i + 1), " ", r.username, " ", r.creditsMinor

File: polyglot/leaderboard_sort.jl
#!/usr/bin/env julia

# Leaderboard sorting (highest balance first).
# Input format: lines like "username creditsMinor"

rows = []
for line in eachline(stdin)
  s = strip(line)
  isempty(s) && continue
  parts = split(s)
  length(parts) < 2 && continue
  push!(rows, (username = parts[1], creditsMinor = parse(Int, parts[2])))
end

sort!(rows, by = r -> r.creditsMinor, rev = true)

for (i, r) in enumerate(rows)
  println(i, " ", r.username, " ", r.creditsMinor)
end

